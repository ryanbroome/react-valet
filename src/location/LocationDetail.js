import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";

import LocationCard from "./LocationCard.js";
import ValetApi from "../api/Api.js";

const LocationDetail = () => {
  const { id } = useParams();
  const [location, setLocation] = useState(null);
  const [loading, setLoading] = useState(true);

  useEffect(
    function fetchLocationAtMount() {
      async function fetchLocation() {
        try {
          const locationRes = await ValetApi.getLocationById(id);
          console.log("LOCATIONRES FOR TESTING ", locationRes.data.location[0]);
          setLocation(locationRes.data.location[0]);
          setLoading(false);
        } catch (err) {
          console.log(err);
        }
      }
      fetchLocation();
    },
    [id]
  );
  if (loading) {
    return (
      <div
        className="spinner-border text-primary"
        role="status">
        <span className="visually-hidden"></span>
      </div>
    );
  } else
    return (
      <div>
        <LocationCard location={location} />
      </div>
    );
};

export default LocationDetail;
