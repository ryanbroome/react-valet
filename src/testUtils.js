import React from "react";
import UserContext from "./auth/UserContext";
// import { render } from "@testing-library/react";
// import { MemoryRouter } from "react-router";
import App from "../src/App";
// import TransactionList from "./TransactionList";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";
  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };
  const mockLocation = {};
  const mockVehicle = {};
  const mockTransaction = {};

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
