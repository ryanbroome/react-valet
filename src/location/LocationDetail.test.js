import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import LocationDetail from "../location/LocationDetail";
import { MemoryRouter } from "react-router";
import { MockUserContext } from "../testUtils";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext children={<LocationDetail />}></MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing", function () {
  render(
    <MemoryRouter initialEntries={["/locations/id/1"]}>
      <MockUserContext children={<LocationDetail />}></MockUserContext>
    </MemoryRouter>
  );
});
