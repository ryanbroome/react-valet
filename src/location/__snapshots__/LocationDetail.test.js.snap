// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetail: matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner-border text-primary"
    role="status"
  >
    <span
      class="visually-hidden"
    />
  </div>
</DocumentFragment>
`;
